packages/react-sdk/src/core/hooks/useTrackElementVisibility.ts

```diff
@@ -13,32 +13,89 @@ export const useTrackElementVisibility = <T extends HTMLElement>({
 }) => {
   const call = useCall();
 
+  const isIntersectingReference = useRef<boolean | null>(null);
+
   const viewportTracker = propsViewportTracker ?? call?.viewportTracker;
 
+  const participantVisibility = useMemo(
+    () => ({
+      update: (isVisible: boolean) => {
+        call?.state.updateParticipant(sessionId, (p) => {
+          // skip update if the participant state is already in the expected state
+          if (
+            (isVisible &&
+              p.viewportVisibilityState === VisibilityState.VISIBLE) ||
+            (!isVisible &&
+              p.viewportVisibilityState === VisibilityState.INVISIBLE)
+          )
+            return p;
+
+          return {
+            ...p,
+            viewportVisibilityState: isVisible
+              ? VisibilityState.VISIBLE
+              : VisibilityState.INVISIBLE,
+          };
+        });
+      },
+      reset: () => {
+        call?.state.updateParticipant(sessionId, (p) => {
+          // skip update if the participant state is already in the expected state
+          if (p.viewportVisibilityState === VisibilityState.UNKNOWN) return p;
+
+          return {
+            ...p,
+            viewportVisibilityState: VisibilityState.UNKNOWN,
+          };
+        });
+      },
+    }),
+    [call, sessionId],
+  );
+
+  useEffect(() => {
+    if (!call || !trackedElement) return;
+
+    const handleVisibilityChange = () => {
+      const isDocumentVisible = document.visibilityState === 'visible';
+      const isPIP = trackedElement.contains(document.pictureInPictureElement);
```

Comment:
What does "PIP" mean? Could you add a comment @arnautov-anton ? I understand that it is picture in picture